### PR TITLE
Feature/kyv 825 mit payments

### PR DESCRIPTION
**Pull request by wpamb**

### CODE REVIEW

File: core/src/main/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateMitChargeRequest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    protected CreateMitChargePayload getPayload(PaytrailClient client) {
+        return payload;
+    }
+
+    @Override
+    protected TreeMap<String, String> getRequestSpecificHeaders() {
+        return null;
```
Comment (severikupari1) [inline core/src/main/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateMitChargeRequest.java:30]: Should transactionId be setted to headers?
https://docs.paytrail.com/#/?id=create-authorization-hold-or-charge 

from paytrail docs:
Note! The transaction id needs to be sent on checkout-transaction-id header as well.

**Review by severikupari1 — COMMENTED** [review]
One question otherwise, very good job!

File: core/src/main/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateMitChargeRequest.java (hunk 1 above)
Comment (wpamb) [reply core/src/main/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateMitChargeRequest.java:30]: But which transactionId? We don't receive a transactionId anywhere in the previous calls. We only receive it upon requesting the mit charge. In the ticket, it's been written that checkout-transaction-id = token. Now, I don't know what's the correct way.

**Review by laukkanenj — APPROVED** [review]

**Review by laukkanenj — APPROVED** [review]

File: core/src/main/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateMitChargeRequest.java (hunk 1 above)
Comment (severikupari1) [reply core/src/main/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateMitChargeRequest.java:30]: This was resolved, token is added to headers in applyDefaultHeaders if checkout-transaction-id is not null :D My bad

**Review by severikupari1 — APPROVED** [review]
Good job!

**Pull request merged**